src/SFA.DAS.EAS.Employer_Financial.Database/Scripts/Manual/GenerateTestData/AdHoc/CreateLevies.sql

```diff
@@ -0,0 +1,146 @@
+-- Instructions: Find clearly signposted 'Levy Generation Knobs' section below to change generation variable defaults
+
+CREATE FUNCTION PayrollMonth (@date datetime)  
+RETURNS int 
+AS  
+BEGIN  
+  declare @month int = DATEPART(month,@date)
+
+  SET @month = @month - 3
+  IF @month < 1
+	SET @month = @month + 12
+
+  RETURN(@month);  
+END; 
+GO
+
+CREATE FUNCTION PayrollYear (@date datetime)  
+RETURNS VARCHAR(5)
+AS  
+BEGIN  
+  declare @month int = DATEPART(month,@date)
+  declare @year int = DATEPART(year,@date)
+
+  if @month < 4
+	SET @year = @year - 1
+	
+  DECLARE @payrollYear VARCHAR(5) = (SELECT RIGHT(CONVERT(VARCHAR(5), @year, 1), 2)) + '-' + (SELECT RIGHT(CONVERT(VARCHAR(4), @year+1, 1), 2))
+  return @payrollYear
+END; 
+GO
+
+BEGIN TRANSACTION CreateLevy
+go
+
+--  __                                   ____                                          __                            __  __                  __                
+-- /\ \                                 /\  _`\                                       /\ \__  __                    /\ \/\ \                /\ \               
+-- \ \ \         __   __  __   __  __   \ \ \L\_\     __    ___      __   _ __    __  \ \ ,_\/\_\    ___     ___    \ \ \/'/'    ___     ___\ \ \____    ____  
+--  \ \ \  __  /'__`\/\ \/\ \ /\ \/\ \   \ \ \L_L   /'__`\/' _ `\  /'__`\/\`'__\/'__`\ \ \ \/\/\ \  / __`\ /' _ `\   \ \ , <   /' _ `\  / __`\ \ '__`\  /',__\ 
+--   \ \ \L\ \/\  __/\ \ \_/ |\ \ \_\ \   \ \ \/, \/\  __//\ \/\ \/\  __/\ \ \//\ \L\.\_\ \ \_\ \ \/\ \L\ \/\ \/\ \   \ \ \\`\ /\ \/\ \/\ \L\ \ \ \L\ \/\__, `\
+--    \ \____/\ \____\\ \___/  \/`____ \   \ \____/\ \____\ \_\ \_\ \____\\ \_\\ \__/.\_\\ \__\\ \_\ \____/\ \_\ \_\   \ \_\ \_\ \_\ \_\ \____/\ \_,__/\/\____/
+--     \/___/  \/____/ \/__/    `/___/> \   \/___/  \/____/\/_/\/_/\/____/ \/_/ \/__/\/_/ \/__/ \/_/\/___/  \/_/\/_/    \/_/\/_/\/_/\/_/\/___/  \/___/  \/___/ 
+--                                 /\___/                                                                                                                      
+--                                 \/__/                                                                                                                       
+
+DECLARE @accountId BIGINT                = 1
```

Comment:
Unfortunately not, as variables don't persist past the end of the batch (defined by the `Go` statements).

That's one of the advantages of the framework scripts - the knobs can be at the top of the scripts, because all the sprocs and functions are created in the separate setup script, whereas the ad-hoc scripts need to set them up _before_ the user-twiddlable knobs.